x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtUtilTests.java

```diff
@@ -44,11 +83,175 @@ public void testSignedJwtGenerateSignVerify() throws Exception {
     private boolean helpTestSignatureAlgorithm(final JWSAlgorithm signatureAlgorithm) throws Exception {
         LOGGER.info("Testing signature algorithm " + signatureAlgorithm);
         // randomSecretOrSecretKeyOrKeyPair() randomizes which JwtUtil methods to call, so it indirectly covers most JwtUtil code
```

Comment:
randomSecretOrSecretKeyOrKeyPair is not here any more, is the "it indirectly covers most JwtUtil code" covered by something else ?

Reply:
I switched to Nimbus JWK and its subclasses. The library is significantly more streamlined if you use JWKs, and there was a lot of missing functionality if using SecretKey/KeyPair objects. These are the 3 main key classes.
- OctetSequenceKey extends JWK implements SecretJWK
- RSAKey RSAKey extends JWK implements AsymmetricJWK
- ECKey extends JWK implements AsymmetricJWK, CurveBasedJWK

Reply:
My comment here was simply questioning the part where "most JwtUtil code is covered" by  "randomSecretOrSecretKeyOrKeyPair" and if we do cover the "JwtUtil" code now that you don't have "randomSecretOrSecretKeyOrKeyPair". This was not about JWK classes vs SecretKey/KeyPair. 

If you think we're covered, I'm more than happy, I was just checking